### PR TITLE
Xeno icons

### PR DESCRIPTION
**Pull request by goose4429**
## About The Pull Request

Adds new minimap icons for xenos! Each tier and caste is now easily seen on the map. And animated silo icon!
All credit to **BOT Purple**, the RUTGMC contributor.

image

image

image

image
image

image

## Why It's Good For The Game

Cool icons are good. 
Animated silo is very good.

## Changelog
:cl:
add: New xeno minimap icons
add: Animated silo icon
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
Very cool.

**Comment by Naaanii** [comment]
Bless.

**Comment by Walarks** [comment]
Georger has new red icon, very based

File: code/controllers/subsystem/minimaps.dm [hunk 1]
```diff
@@ -201,13 +201,19 @@ SUBSYSTEM_DEF(minimaps)
  * * iconstate: iconstate for the blip we want to be used for this tracked atom
  * * icon: icon file we want to use for this blip, 'icons/UI_icons/map_blips.dmi' by efault
  */
-/datum/controller/subsystem/minimaps/proc/add_marker(atom/target, zlevel, hud_flags = NONE, iconstate, icon = 'icons/UI_icons/map_blips.dmi')
+/datum/controller/subsystem/minimaps/proc/add_marker(atom/target, zlevel, hud_flags = NONE, iconstate, icon = 'icons/UI_icons/map_blips.dmi', overlay_iconstates = null)
```
Comment (TiviPlus) [inline code/controllers/subsystem/minimaps.dm:204]:
```suggestion
/datum/controller/subsystem/minimaps/proc/add_marker(atom/target, zlevel, hud_flags = NONE, iconstate, icon = 'icons/UI_icons/map_blips.dmi', list/overlay_iconstates)
```

File: code/controllers/subsystem/minimaps.dm [hunk 2]
```diff
@@ -201,13 +201,19 @@ SUBSYSTEM_DEF(minimaps)
...
 		CRASH("Invalid marker added to subsystem")
 	if(!initialized)
 		earlyadds += CALLBACK(src, .proc/add_marker, target, zlevel, hud_flags, iconstate, icon)
 		return
+
 	var/image/blip = image(icon, iconstate, pixel_x = MINIMAP_PIXEL_FROM_WORLD(target.x) + minimaps_by_z["[zlevel]"].x_offset, pixel_y = MINIMAP_PIXEL_FROM_WORLD(target.y) + minimaps_by_z["[zlevel]"].y_offset)
+
+	if (overlay_iconstates)
+		for(var/i in overlay_iconstates)
+			blip.overlays += image(icon, i)
```
Comment (TiviPlus) [inline code/controllers/subsystem/minimaps.dm:215]:
```suggestion
	for(var/i in overlay_iconstates)
		blip.overlays += image(icon, i)
```

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 3]
```diff
@@ -688,3 +688,14 @@
 ///Handles empowered abilities, should return TRUE if the ability should be empowered. Empowerable should be FALSE if the ability cannot itself be empowered but has interactions with empowerable abilities
 /mob/living/carbon/xenomorph/proc/empower(empowerable = TRUE)
 	return FALSE
+
+/mob/living/carbon/xenomorph/proc/update_leader_icon(isleader)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:692]: no autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Losenis** [comment]
Jesus christ change the color from red to something else noticeable, the contrast is so low when it's downsized to actual minimap size that you can't tell them apart

File: code/controllers/subsystem/minimaps.dm [hunk 4]
```diff
@@ -201,13 +201,18 @@ SUBSYSTEM_DEF(minimaps)
  * * iconstate: iconstate for the blip we want to be used for this tracked atom
  * * icon: icon file we want to use for this blip, 'icons/UI_icons/map_blips.dmi' by efault
  */
-/datum/controller/subsystem/minimaps/proc/add_marker(atom/target, zlevel, hud_flags = NONE, iconstate, icon = 'icons/UI_icons/map_blips.dmi')
+/datum/controller/subsystem/minimaps/proc/add_marker(atom/target, zlevel, hud_flags = NONE, iconstate, icon = 'icons/UI_icons/map_blips.dmi', list/overlay_iconstates)
```
Comment (magatsuchi) [inline code/controllers/subsystem/minimaps.dm:204]: please document your new arguement in the documentation above

**Review by magatsuchi — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
@Kuro020

**Review by Kuro020 — APPROVED** [review]

**Comment by Wayland-Smithy** [comment]
Not sure if you fixed this yet, but both larva and hivemind (its physical form not the core) appear as the `xenominion` icon.

**Pull request merged**